src/main/java/at/hannibal2/skyhanni/api/HotmAPI.kt

```diff
@@ -0,0 +1,113 @@
+
+object HotmAPI {
+
+    fun copyCurrentTree() = HotmData.storage?.deepCopy()
+
+    val activeMiningAbility get() = HotmData.abilities.firstOrNull { it.enabled }
+
+    private val blueGoblinEgg = "goblin_omelette_blue_cheese".asInternalName()
+
+    private val blueEggCache = TimeLimitedCache<ItemStack, Boolean>(10.0.seconds)
+    val isBlueEggActive
+        get() = InventoryUtils.getItemInHand()?.let {
+            blueEggCache.getOrPut(it) {
+                it.getItemCategoryOrNull() == ItemCategory.DRILL && it.getDrillUpgrades()
+                    ?.contains(blueGoblinEgg) == true
+            }
+        } == true
+
+    enum class Powder() {
+        MITHRIL,
+        GEMSTONE,
+        GLACITE,
+
+        ;
+
+        val lowName = name.lowercase().firstLetterUppercase()
+
+        val heartPattern by RepoPattern.pattern(
+            "inventory.${name.lowercase()}.heart",
+            "§7$lowName Powder: §a§.(?<powder>[\\d,]+)"
+        )
+        val resetPattern by RepoPattern.pattern(
+            "inventory.${name.lowercase()}.reset",
+            "\\s+§8- §.(?<powder>[\\d,]+) $lowName Powder"
+        )
+
+        fun pattern(isHeart: Boolean) = if (isHeart) heartPattern else resetPattern
+
+        fun getStorage() = ProfileStorageData.profileSpecific?.mining?.powder?.get(this)
+
+        fun getCurrent() = getStorage()?.available ?: 0L
+
+        fun setCurrent(value: Long) {
+            getStorage()?.available = value
+        }
+
+        fun addCurrent(value: Long) {
+            setCurrent(getCurrent() + value)
+        }
+
+        fun getTotal() = getStorage()?.total ?: 0L
+
+        fun setTotal(value: Long) {
+            getStorage()?.total = value
+        }
+
+        fun addTotal(value: Long) {
+            setTotal(getTotal() + value)
+        }
+
+        /** Use when new powder gets collected*/
+        fun gain(value: Long) {
+            addTotal(value)
+            addCurrent(value)
+        }
+
+        fun reset() {
+            setCurrent(0)
+            setTotal(0)
+        }
+    }
+
+    var skymall: SkymallPerk? = null
+
+    var mineshaftMayhem: MayhemPerk? = null
+
+    enum class SkymallPerk(chat: String, itemString: String) {
+        MINING_SPEED("Gain §r§a+100 §r§6⸕ Mining Speed§r§f.", "Gain §a+100 §6⸕ Mining Speed§7."),
+        MINING_FORTUNE("Gain §r§a+50 §r§6☘ Mining Fortune§r§f.", "Gain §a+50 §6☘ Mining Fortune§7."),
+        EXTRA_POWDER("Gain §r§a+15% §r§fmore Powder while mining.", "Gain §a+15% §7more Powder while mining."),
+        ABILITY_COOLDOWN("Reduce Pickaxe Ability cooldown by §r§a20%§r§f.", "Reduce Pickaxe Ability cooldown by"),
+        GOBLIN_CHANCE("§r§a10x §r§fchance to find Golden and Diamond Goblins.", "§a10x §7chance to find Golden and"),
+        TITANIUM("Gain §r§a5x §r§9Titanium §r§fdrops", "Gain §a+15% §7more Powder while mining.")
+        ;
+
+        val chatPattern by RepoPattern.pattern("mining.hotm.skymall.chat.$name", chat)
```

Comment:
Crash on startup. replace `_` with `.`, and lowercase
```
Caused by: java.lang.IllegalArgumentException: pattern key: "mining.hotm.skymall.chat.MINING_SPEED" failed shape requirements
at at.hannibal2.skyhanni.utils.repopatterns.RepoPatternManager.verifyKeyShape(RepoPatternManager.kt:137)
at at.hannibal2.skyhanni.utils.repopatterns.RepoPatternManager.of(RepoPatternManager.kt:168)
at at.hannibal2.skyhanni.utils.repopatterns.RepoPattern$Companion.pattern(RepoPattern.kt:94)
at at.hannibal2.skyhanni.api.HotmAPI$SkymallPerk.<init>(HotmAPI.kt:100)
at at.hannibal2.skyhanni.api.HotmAPI$SkymallPerk.<clinit>(HotmAPI.kt:92)
```

Reply:
fix it.